### ISSUE TITLE
Add to MELPA

### ISSUE THREAD

**Issue by mkleehammer** [Overdr0ne/gumshoe#2]
Are there plans to add this to MELPA?  Thanks.

**Comment by Overdr0ne** [comment]
I've never added a package to melpa before, but I don't see why not. I'll look into it in the next few days when I get the chance.

**Comment by Overdr0ne** [comment]
Hey, just to update, I decided to add support for perspective-local and buffer-local tracking. I want to finish that and get the interfaces and package boundaries all nailed down before integrating and declaring some kind of version 1.0. It was a bit more complicated than I was expecting, but things are mostly working now. I expect to be finished soon...

**Comment by Overdr0ne** [comment]
I expect it will be integrated sometime in the next few weeks. I'll update then...
https://github.com/melpa/melpa/pull/7662

**Comment by minad** [comment]
Great to see that your package will soon land in MELPA! I just looked at the code again and it is great that you are continuously improving the package. One thing I wondered about - why do you use eieio? I haven't used eieio myself yet - but it seems you don't rely on oop features?

**Comment by Overdr0ne** [comment]
> One thing I wondered about - why do you use eieio? I haven't used eieio myself yet - but it seems you don't rely on oop features?

Good question. tl;dr it's a relic from an old design

When I had originally begun implementing scoped logging, I'd intended to use multiple dispatch to handle each scope differently. Basically, I wanted to contain everything for a mode of some scope into one object, then build a common interface for all of them. After thinking about it, I realized I could just use emacs' built-in scoping rules for mostly the same purpose, by writing common macros operating on symbols, and resolving those symbols at runtime to the object of the correct scope. I had also toyed with the idea of using mutliple dispatch to make a sort of fake namespace by passing around a 'gumshoe' object, but my attempts to generate method signatures the help system would be able to differentiate didn't really work out. I'm conflicted, I like eieio, mutliple inheritance aside, but it's just not very well integrated with the rest of the system. And at this point I could definitely just use structs instead, and probably should to get rid of that dependency.
